src/main/webapp/WEB-INF/tags/instructor/comments/visibilityOptions.tag

```diff
@@ -0,0 +1,231 @@
+                                ${commentRow.visibilityCheckboxes.recipientSectionAbleToSeeComment ? "checked=\"checked\"" : ""}>
+                        </td>
+                        <td>
+                            <input 
+                                class="visibilityCheckbox giverCheckbox"
+                                type="checkbox"
+                                value="<%= CommentParticipantType.SECTION %>"
+                                ${commentRow.visibilityCheckboxes.recipientSectionAbleToSeeGiverName ? "checked=\"checked\"" : ""}>
+                        </td>
+                        <td>
+                            <input 
+                                class="visibilityCheckbox recipientCheckbox"
+                                type="checkbox"
+                                value="<%= CommentParticipantType.SECTION %>"
+                                ${commentRow.comment.recipientType == 'SECTION' ? "disabled=\"disabled\"" : ""}
+                                ${commentRow.visibilityCheckboxes.recipientSectionAbleToSeeRecipientName ? "checked=\"checked\"" : ""}>
+                        </td>
+                    </tr>
+                </c:if>
+                <tr id="recipient-course${commentIdx}">
+                    <td class="text-left">
+                        <div data-toggle="tooltip"
+                            data-placement="top" title=""
+                            data-original-title="Control what other students in this course can view">
+                            ${commentRow.comment.recipientType == 'COURSE' ? "Students in this course" : "Other students in this course"}</div>
+                    </td>
+                    <td>
+                        <input 
+                            class="visibilityCheckbox answerCheckbox"
+                            type="checkbox" value="<%= CommentParticipantType.COURSE %>"
+                            ${commentRow.visibilityCheckboxes.courseStudentsAbleToSeeComment ? "checked=\"checked\"" : ""}>
+                    </td>
+                    <td>
+                        <input 
+                            class="visibilityCheckbox giverCheckbox"
+                            type="checkbox" value="<%= CommentParticipantType.COURSE %>"
+                            ${commentRow.visibilityCheckboxes.courseStudentsAbleToSeeGiverName ? "checked=\"checked\"" : ""}>
+                    </td>
+                    <td>
+                        <input 
+                            class="visibilityCheckbox recipientCheckbox"
+                            type="checkbox" value="<%= CommentParticipantType.COURSE %>"
+                            ${commentRow.comment.recipientType == 'COURSE' ? "disabled=\"disabled\"" : ""}
+                            ${commentRow.visibilityCheckboxes.courseStudentsAbleToSeeRecipientName ? "checked=\"checked\"" : ""}>
+                    </td>

```

Comment:
Wait, actually the visibility checkbox for Comment and FeedbackResponseComment are different. So if you're working with IFResults it should be independent of this.
Meanwhile, mine (ISRecords) be dependent on both :v: :v: